sec_certs_page/common/bindings.py

```diff
@@ -0,0 +1,510 @@
+                f"Copied {len(os.listdir(path))} {file_ext} files from {path} to {output_dir}.")
+    else:
+        click.echo(f"Error: Unsupported URL format of: {url}")
+
+
+def download_all_bindings_sync(
+        urls: list[str],
+        output_dir: str,
+        verbose: bool = False) -> None:
+    """ Download binding files from all given URLs
+
+    Args:
+        urls (list[str]): list of URLs to download from
+        output_dir (str): Path to write the binding files
+        verbose (bool, optional): Whether to click.echo out extra information. Defaults to False.
+    """
+    os.makedirs(output_dir, exist_ok=True)
+
+    for url in urls:
+        asyncio.run(
+            download_binding_files(
+                url,
+                output_dir,
+                verbose=verbose))
+        if verbose:
+            click.echo(f"Download completed for {url}.\n")
+
+
+async def process_binding_files(bindings_dir: str, download_dir: str, download_headers: bool = True, verbose: bool = True) -> list[Binding]:
+    """ Processing of all binding files in given directory
+
+    Args:
+        bindings_dir (str): Directory where binding files are located
+        download_dir (str): Directory where binding files are to be downloaded
+        download_headers (bool, optional): Whether to download the headers files. Defaults to True.
+        verbose (bool, optional): _description_. Defaults to True.
+
+    Returns:
+        list[Binding]: list of binding Objects identified in the binding files
+    """
+    os.makedirs(download_dir, exist_ok=True)
+    tasks = []
+    bindings = []
+    for filename in os.listdir(bindings_dir):
+        if filename.endswith(".json"):
+            if verbose:
+                click.echo(f"\nProcessing binding file: {filename}")
+            filepath = os.path.join(bindings_dir, filename)
+            with open(filepath, "r", encoding="utf8") as f:
+                data = json.load(f)
+                if not verify_jwt(data):
+                    click.echo(
+                        f"JWT verification failed processing binding file {filename}")
+                    continue
+                for item in data["data"]:
+                    timestamp = item["timestamp"]
+                    if not verify_timestamp(timestamp):
+                        click.echo(f"Invalid timestamp for file: {filename}")
+                        continue
+                    url = item["metadata_header_url"]
+                    header_file_name = get_filename_from_url(
+                        urllib.parse.unquote(url))
+                    if not verify_url(url):
+                        click.echo(
+                            f"Invalid metadata_header_url for file: {url}")
+                        continue
+                    if not download_headers:
+                        continue
+
+                    for cert_id in item["certificate_ids"]:
+                        bindings.append(Binding(cert_id, header_file_name))
```

Comment:
These lines should be flipped right? the check should come after the items are added to the bindings list.